File: db4oj/core/src/com/db4o/typehandlers/SingleClassTypeHandlerPredicate.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */

package com.db4o.typehandlers;

import com.db4o.reflect.*;

/**
 * allows installing a Typehandler for a single class.
 */
public final class SingleClassTypeHandlerPredicate implements TypeHandlerPredicate {
    
	private final Class _class;
	
	public SingleClassTypeHandlerPredicate(Class clazz) {
		_class = clazz;
	}
	
	public boolean match(ReflectClass candidate) {
		final ReflectClass reflectClass = candidate.reflector().forClass(_class);
		return candidate == reflectClass;
	}
}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Typehandlers/SingleClassTypeHandlerPredicate.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using System;
using Db4objects.Db4o.Reflect;
using Db4objects.Db4o.Typehandlers;

namespace Db4objects.Db4o.Typehandlers
{
	/// <summary>allows installing a Typehandler for a single class.</summary>
	/// <remarks>allows installing a Typehandler for a single class.</remarks>
	public sealed class SingleClassTypeHandlerPredicate : ITypeHandlerPredicate
	{
		private readonly Type _class;

		public SingleClassTypeHandlerPredicate(Type clazz)
		{
			_class = clazz;
		}

		public bool Match(IReflectClass candidate)
		{
			IReflectClass reflectClass = candidate.Reflector().ForClass(_class);
			return candidate == reflectClass;
		}
	}
}
